opal/mca/pmix/pmix112/pmix/src/usock/usock_sendrecv.c

```diff
@@ -99,15 +101,22 @@ static void lost_connection(pmix_peer_t *peer, pmix_status_t err)
                 }
             }
         }
-         /* remove this proc from the list of ranks for this nspace */
-         pmix_list_remove_item(&(peer->info->nptr->server->ranks), &(peer->info->super));
-         PMIX_RELEASE(peer->info);
-         /* reduce the number of local procs */
-         --peer->info->nptr->server->nlocalprocs;
-         /* do some cleanup as the client has left us */
-         pmix_pointer_array_set_item(&pmix_server_globals.clients,
+        /* remove this proc from the list of ranks for this nspace */
+        pmix_list_remove_item(&(peer->info->nptr->server->ranks), &(peer->info->super));
+        PMIX_RELEASE(peer->info);
+        /* reduce the number of local procs */
+        --peer->info->nptr->server->nlocalprocs;
+        /* do some cleanup as the client has left us */
+        pmix_pointer_array_set_item(&pmix_server_globals.clients,
                                      peer->index, NULL);
-         PMIX_RELEASE(peer);
+        /* remove all registered event handlers so libevent doesn't complain */
+        PMIX_LIST_FOREACH_SAFE(reginfoptr, regnext, &pmix_server_globals.client_eventregs, pmix_regevents_info_t) {
+            if (reginfoptr->peer == peer) {
+                PMIX_RELEASE(reginfoptr);
```

Comment:
Per IM discussion -- @rhc54 is going to bring over another change here for removing the item from the list.